.github/workflows/build-and-push-docker-hub.yaml

```diff
@@ -0,0 +1,35 @@
+name: Build and Push Docker Image to Docker Hub
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+  workflow_dispatch:
+
+jobs:
+  set_env:
+    name: Create version tag
+    runs-on: ubuntu-latest
+    outputs:
+      tags: ${{ steps.version_step.outputs.tags }}
+      zallet_version: ${{ steps.version_step.outputs.zallet_version }}
+    steps:
+    - id: version_step
+      run:  |
+        echo "tags=latest,${{ github.ref_name }},${{ github.sha }}" >> $GITHUB_OUTPUT
+        echo "zallet_version=$(echo ${{ github.ref_name }} | sed 's/v//g')" >> $GITHUB_OUTPUT
+
+  build_push:
+    uses: zcash/.github/.github/workflows/build-and-push-docker-hub.yaml@main
```

Comment:
What is this doing? It looks recursive to me which makes no sense. Is this actually pulling from the `zcash/zcash` repo?